server/src/main/java/org/exoplatform/chat/services/mongodb/MongoModule.java

```diff
@@ -15,5 +19,21 @@ protected void configure() {
     bind(NotificationService.class).to(NotificationServiceImpl.class);
     bind(TokenService.class).to(TokenServiceImpl.class);
     bind(UserService.class).to(UserServiceImpl.class);
+    bind(IdentityManager.class).toProvider(getProvider(IdentityManager.class));
+  }
+
+  protected <T> Provider<T> getProvider(final Class<T> clazz) {
+    return new Provider<T>() {
+      @Override
+      public T get() {
+        RootContainer rootContainer = RootContainer.getInstance();
+        T ret = rootContainer.getComponentInstanceOfType(clazz);
+        if(ret == null) {
+          PortalContainer portalContainer = PortalContainer.getInstance();
+          ret = portalContainer.getComponentInstanceOfType(clazz);
+        }
+        return ret;
+      }
+    };
```

Comment:
IdentityManager is only used in UserServiceImpl.java, so just move these code there for now.